hack/boilerplate/boilerplate.py

```diff
@@ -16,22 +16,46 @@
 
 from __future__ import print_function
 
+import argparse
 import json
 import mmap
 import os
 import re
 import sys
 
-def PrintError(*err):
-  print(*err, file=sys.stderr)
+parser = argparse.ArgumentParser()
+parser.add_argument("filenames", help="list of files to check, all files if unspecified", nargs='*')
+args = parser.parse_args()
 
-def file_passes(filename, extension, ref, regexs):
+rootdir = os.path.dirname(__file__) + "/../../"
+rootdir = os.path.abspath(rootdir)
+
+def get_ref(extension, refs):
+    if extension in refs:
```

Comment:
probably not. this came out of the 'old' way where we were only loading the single reference file.  The cost of just loading all 3 of them up front is so minuscule and it probably would make the code more readable.